client/components/addworkout.jsx

```diff
@@ -36,24 +36,32 @@ export default class AddWorkout extends React.Component {
     this.showEditModal = this.showEditModal.bind(this);
     this.getWorkout = this.getWorkout.bind(this);
     this.getCompleted = this.getCompleted.bind(this);
+    this.getCompletedDates = this.getCompletedDates.bind(this);
     this.getEdit = this.getEdit.bind(this);
     this.deleteExercise = this.deleteExercise.bind(this);
     this.mapExercise = this.mapExercise.bind(this);
     this.showWorkout = this.showWorkout.bind(this);
     this.toggleComplete = this.toggleComplete.bind(this);
   }
 
-  // when it mounts, fetch data and show for specific date
   componentDidMount() {
     this.getWorkout();
     this.getCompleted();
+    this.getCompletedDates();
   }
 
   toggleAddModal() {
-    this.setState({ addModalActive: !this.state.addModalActive });
+    this.setState({ addModalActive: !this.state.addModalActive }, () => {
+      if (!this.state.addModalActive) {
+        this.setState({ exercise: '' });
+        this.setState({ weight: '' });
+        this.setState({ sets: '' });
+        this.setState({ reps: '' });
+        this.setState({ rest: '' });
```

Comment:
You should do this with a single `setState` call, just pass it a single object with all the properties you wish to set